src/push.go

```diff
@@ -53,7 +58,54 @@ func (f *PushOnlyFlags) Validate() Validations {
 	return validations
 }
 
+func GetToken(ctx context.Context, flags *PushFlags) (*string, error) {
+	fmt.Printf("Getting an impersonation token for `%s`..\n", flags.ActionsAdminUser)
+
+	ts := oauth2.StaticTokenSource(&oauth2.Token{AccessToken: flags.Token})
+	tc := oauth2.NewClient(ctx, ts)
+	ghClient, err := github.NewEnterpriseClient(flags.BaseURL, flags.BaseURL, tc)
+	if err != nil {
+		return nil, errors.Wrap(err, "error creating enterprise client")
+	}
+
+	rootRequest, err := ghClient.NewRequest("GET", enterpriseAPIPath, nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "Error constructing request for GitHub Enterprise client.")
+	}
+	rootResponse, err := ghClient.Do(ctx, rootRequest, nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "Error checking connectivity for GitHub Enterprise client.")
+	}
+
+	isAE := rootResponse.Header.Get(enterpriseVersionHeaderKey) == enterpriseAegisVersionHeaderValue
+
+	minimumRepositoryScope := "public_repo"
+	if isAE {
+		// the default repository scope for non-ae instances is 'public_repo'
+		// while it is `repo` for ae.
+		minimumRepositoryScope = "repo"
+		fmt.Printf("Running against GitHub AE, changing the repository scope to '%s'..\n", minimumRepositoryScope)
+	}
+
+	impersonationToken, _, err := ghClient.Admin.CreateUserImpersonation(ctx, flags.ActionsAdminUser, &github.ImpersonateUserOptions{Scopes: []string{minimumRepositoryScope, "workflow"}})
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to impersonate Actions admin user.")
+	}
+
+	fmt.Printf("Got the impersonation token for `%s`..\n", flags.ActionsAdminUser)
+	var result = impersonationToken.GetToken()
+	return &result, nil
```

Comment:
The reason to use `*string` here was because returning `""` for the error felt.. weird. And I can't do `nil, error` if the type is `string`.
I would never use the pointer otherwise.
Or is returning `""` more idiomatic?

Reply:
It is fairly common in Go to return the zero-value for the particular type if an error occurs. It feels a bit weird to me too, but I would say it's idiomatic.

Reply:
Agreed, returning empty string is idiomatic since the caller (should) always check that the error value is not-nil before doing anything with the other return values.

Reply:
Alrighty, thank you for the suggestion @jidicula @chrisgavin!
Changed the type to `string`.